plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -477,15 +479,15 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         // Extract identity partition fields that are present in all partition specs, for creating the discrete predicates.
         Set<Integer> partitionSourceIds = identityPartitionColumnsInAllSpecs(icebergTable);
 
+        // Extract identity partition columns
+        Map<Integer, IcebergColumnHandle> columns = getColumns(icebergTable.schema(), typeManager).stream()
+                .filter(column -> partitionSourceIds.contains(column.getId()))
+                .collect(toImmutableMap(IcebergColumnHandle::getId, identity()));
+
         TupleDomain<IcebergColumnHandle> enforcedPredicate = table.getEnforcedPredicate();
 
         DiscretePredicates discretePredicates = null;
-        if (!partitionSourceIds.isEmpty()) {
-            // Extract identity partition columns
-            Map<Integer, IcebergColumnHandle> columns = getColumns(icebergTable.schema(), typeManager).stream()
-                    .filter(column -> partitionSourceIds.contains(column.getId()))
-                    .collect(toImmutableMap(IcebergColumnHandle::getId, identity()));
-
+        if (!partitionSourceIds.isEmpty() && !columns.isEmpty()) {
```

Comment:
I find the current solution a bit hard to understand just by reading the code. It is rather unclear why the `columns` is supposed to be empty when `partitionSourceIds` is not empty. 
Maybe adding an extra comment would help?

Unrelated to the current PR: When Trino is going to be able to deal with nested fields as partitions is the output of `getColumns(...)` method supposed to change?

Reply:
> why the columns is supposed to be empty when partitionSourceIds is not empty.

Do you mean "columns is **_not_** supposed to be empty when `partitionSourceIds` is not empty"?

Yes, the method `getColumns` will be changed when we support nested fields as partitions.

Reply:
> Do you mean "columns is not supposed to be empty when partitionSourceIds is not empty"?

I mean that something that would hint a code maintainer that the partition source ids may point to a nested field which will not be returned as a column handle may by `getColumns(..)` may help.

Reply:
I think the existing name `columns` is just not descriptive. Maybe something like `identityPartitionColumns`?

It also may be a redundant check. If `partitionSourceIds` is not empty it's impossible for `columns` to be empty. So the second block of the if here could be a `verify` instead.

Reply:
>  If partitionSourceIds is not empty it's impossible for columns to be empty.

No, that's possible in case of nested fields.